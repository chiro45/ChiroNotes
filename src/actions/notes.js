import Swal from "sweetalert2";


import { db } from "../firebase/firebaseConfing";

import { types } from "../types/types";

import {loadNotes} from '../helpers/loadNotes'
import { fileUpload } from "../helpers/fileUpload";




export const startNewNote = ()=>{
    return async (dispatch, getState)=>{       //el getState nos trae todo el state de la aplicacion
        const {uid} = getState().auth;   
        const newNote = {
            title: '',
            body: '',
            date: new Date().getTime()
        }
        const doc =  await db.collection(`${uid}/journal/notes`).add(newNote)     
        
        dispatch(activeNote(doc.id, newNote))

        dispatch(addNewNote(doc.id, newNote))
    }
}

export const activeNote = (id,note )=>({
        type: types.notesActive,
        payload: {
            id,
            ...note
        }
});

export const addNewNote = (id, note)=>({
        type: types.notesAddNew,
        payload:{
            id,
            ...note
        }
})


export const startLoadingNotes = (uid) =>{
    return async (dispatch)=>{
        const notes = await loadNotes(uid)
      
        dispatch(setNotes(notes))
    }
}
export const setNotes = (notes)=>({
    type: types.notesLoad,
    payload: notes
})


 export const startSaveNote = (note)=>{
    return async(dispatch, getState)=>{
        try {
            const {uid} = getState().auth;
        if(!note.url){
            delete note.url;
        }
        const noteToFirestore ={...note}

        delete noteToFirestore.id;

        if(note.title.length > 1){
        await db.doc(`${uid}/journal/notes/${note.id}`).update(noteToFirestore);

        dispatch(refreshNote(note.id,note ))

        Swal.fire('Guardado!', note.title, "success")
        }else{
            Swal.fire('Deberia tener un titulo!', note.title, "error")
        }

        } catch (error) {
            Swal.fire('Error en el guardado!',"hubo un problema, intenta nuevamente", "error")
        }


    }
}

export const refreshNote = (id, note) =>({
    type: types.notesUpdated,
    payload:{
        id,
        note:{
                id,
                ...note
        }

        
    }
})


export const startUploading =(file) =>{
    return async(dispatch, getState)=>{
        const {active:activeNote} = getState().notes

        Swal.fire({
            title: 'Subiendo...',
            text: 'Por Favor Espere...',
            allowOutsideClick: false,
            didOpen:()=>{
                Swal.showLoading()
            }
            
        });
       
        const fileUrl = await fileUpload(file)
        activeNote.url = fileUrl;
        dispatch(startSaveNote(activeNote))

        Swal.close();
    }
}


export const startDeleting = (id)=>{
    return async(dispatch, getState)=>{
        
        const uid = getState().auth.uid;

        await db.doc(`${uid}/journal/notes/${id}`).delete();

        dispatch(deleteNote(id))

        Swal.fire('Eliminado!','', "success")
    }
}


export const deleteNote = (id)=>({
    type: types.notesDelete,
    payload: id
})


export const noteLogout = () =>({
    type: types.notesLogoutCleaning
   
})
