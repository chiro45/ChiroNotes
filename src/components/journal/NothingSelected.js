

export const NothingSelected = () => {
  return (
    <div className="nothing__main-content ">

        <p>Selecciona Alguna Nota O Cree Una Nota</p>
       

        <i className=" far fa-edit fa-4x marginTop-5"></i>

    </div>
  )
  
};
