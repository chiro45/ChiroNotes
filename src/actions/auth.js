import Swal from 'sweetalert2'


import { types } from "../types/types"
import {firebase, googleAuthProvider} from '../firebase/firebaseConfing'
import { finishLoading, startLoading } from "./ui"
import { noteLogout} from './notes'
//acciones que son usadas en el auth




export const startLoginEmailPassword = (email,password)=>{
    return(dispatch)=>{
        
        dispatch(startLoading())
        firebase.auth().signInWithEmailAndPassword(email,password)
        .then( ({ user }) =>{


            dispatch(login(user.uid, user.displayName))
            
            
        dispatch(finishLoading())

        })
        .catch(e => {
        Swal.fire("Error","El usuario o contraseÃ±a son incorrectos","error")
    })
}}
export const startRegisterWhithEmailPasswordName = (email,password, name)=>{
    return(dispatch)=>{
        firebase.auth().createUserWithEmailAndPassword(email,password)
        .then( async({ user })=>{

            await user.updateProfile({
                displayName: name

            })

            dispatch(
                login(user.uid, user.displayName)
             )
            
         
        })
        .catch(e => {
            Swal.fire("Error","El usuario ya ha sido registrado","error")
        
    })

}}


export const startGoogleLogin = ()=>{
    return (dispatch)=>{
       firebase.auth().signInWithPopup(googleAuthProvider) 
       .then(({ user })=>{

           dispatch(
               login(user.uid, user.displayName, user.photoURL),

               console.log(user.photoURL)
                 
               )
            })
            
       
       
       
    }
}


export const login = (uid, displayName, photoURL)=>{
        return{
            type: types.login,
            payload: {
                uid,
                photoURL,
                displayName,
            }
        }
}


export const startLogout = () =>{
    return async(dispatch)=>{
        firebase.auth().signOut()

        dispatch(logout())
        
        dispatch(noteLogout())
    }
}

export const logout = () =>({
    type: types.logout
})


