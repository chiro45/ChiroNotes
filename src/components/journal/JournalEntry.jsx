import React from 'react';
import moment from 'moment'
import PropTypes from 'prop-types';


import 'moment/locale/es'
import { useDispatch } from 'react-redux';
import { activeNote } from '../../actions/notes';

export const JournalEntry = ({id,date,title,body,url}) => {
    const dispatch = useDispatch()
    const noteDate = moment(date);
    //click event sobre alguna de las notas
    const handleEntryClick = ()=>{

        //realizamos el disparo mandando el id y los demas datos de la nota
        dispatch(activeNote(id,{
            date,title,body,url
        })

        );
    }
   
    return (
      <div className='journal__entry pointer animate__animated animate__fadeInRight animate__fast'
      onClick={handleEntryClick}>
      { //en el caso de que la nota tenga imagen entonces se muestra el div
          url &&
      <div className='journal__entry-picture'
      style={{
          backgroundSize:'cover',
          backgroundImage:`url(${url})`,
      }}>
              
      </div>
    }
        
        <div className='journal__entry-body'>
        <p className='journal__entry-title'>
            {title}
        </p>
        <p className='journal__entry-content'>
            {body}
        </p>
        </div>

        <div className='journal__entry-date-box'>
            <span>{noteDate.format('dddd')}</span>
            <h4>{noteDate.format('Do')}</h4>
        </div>
      </div>
  )
};



JournalEntry.propTypes ={
    id: PropTypes.string,
    date: PropTypes.number,
    title: PropTypes.string,
    body: PropTypes.string,
    url: PropTypes.string
}