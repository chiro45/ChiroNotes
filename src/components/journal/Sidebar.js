
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { startLogout } from '../../actions/auth';
import { startNewNote } from '../../actions/notes';
import { JournalEntries } from './JournalEntries';




export const Sidebar = () => {

  const dispatch = useDispatch();
  const {name} = useSelector(state => state.auth);
  
  const handleLogout=()=>{

    dispatch(startLogout())
  }


  const handleAddNew = ()=>{
    dispatch(startNewNote())
    
  }
  
  return (
  <aside className='journal__sidebar '>

      <div className='journal__sidebar-navbar '>
          <h3 className='marginTop-5'>
          
          <span className='journal__name'><i className="fas fa-user"></i> {name}</span>
          </h3>
      </div>

      <div className='journal__new-entry'
      onClick={handleAddNew}>
            <i className='far fa-calendar-plus fa-5x'></i>
            <p className='marginTop-5'> Nueva Entrada</p>
      </div>
      <JournalEntries />
        <button 
        className='btn-navBar'
        onClick={handleLogout}
        >
            Cerrar SesiÃ³n <i className="fas fa-sign-out-alt"></i>
        </button>
  </aside>
  )
};
