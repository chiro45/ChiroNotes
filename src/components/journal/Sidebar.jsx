
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { startLogout } from '../../actions/auth';
import { startNewNote } from '../../actions/notes';
import { JournalEntries } from './JournalEntries';




export const Sidebar = () => {
 // declaramos el dispatch del redux
  const dispatch = useDispatch();
  //extraemos el name del stores del state.auth
  const {name , photoURL} = useSelector(state => state.auth);


  const state = useSelector(state => state);
 
  
 
  
  const handleLogout=()=>{

    dispatch(startLogout())
  }

  //activa el disparo del tartnNewNote
  const handleAddNew = ()=>{
    dispatch(startNewNote())
    
  }
  
  return (
  <aside className='journal__sidebar ' id='sidebar'>

      <div className='journal__sidebar-navbar '>
          <div className='journal__containerName'>
          
          {
             (photoURL)
              ? <div >  <img className='journal__img' src={photoURL}/> <p className='journal__name'>{name}</p></div>
              : <div >  <i className=" journal__img fas fa-user"></i> <p className='journal__name'>{name}</p></div>
              // : <span className='journal__name'>   {name}</span>
          }
          </div>
      </div>

      <div className='journal__new-entry'
      onClick={handleAddNew}>
            <i className='far fa-calendar-plus fa-5x'></i>
            <p className='marginTop-5'> Nueva Entrada</p>
      </div>
      <JournalEntries />
        <button 
        className='btn-navBar'
        onClick={handleLogout}
        >
            Cerrar Sesi√≥n <i className="fas fa-sign-out-alt"></i>
        </button>
  </aside>
  )
};
