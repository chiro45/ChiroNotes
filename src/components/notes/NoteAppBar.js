import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { startSaveNote, startUploading } from '../../actions/notes';
import moment from 'moment'
export const NoteAppBar = ({date}) => {
    const {active:note} = useSelector(state => state.notes)
    const dispatch = useDispatch();

    const handleSave =()=>{
        dispatch(startSaveNote(note))
    }

    const handlePictureClick= () =>{
        document.getElementById('fileSelector').click()




    }


    const handleFileChange = (e)=>{
            const file = e.target.files[0];
            if (file){
                dispatch(startUploading(file))
            }
    }

    
    
  return (
    <div className='notes__appbar'>
  
        <span>{moment().subtract(10, 'days').calendar()}</span>
        <input id='fileSelector'
               type='file'
               name='file'
               style={{display: 'none'}}
               onChange={handleFileChange}
        />
        <div>
            <button className='btn'
            onClick={handlePictureClick}>Foto</button>
        
            <button className='btn'
            onClick={handleSave}>Guardar</button>
        </div>
    

    </div>)
};
