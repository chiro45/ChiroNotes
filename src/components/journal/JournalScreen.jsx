
import { NothingSelected } from './NothingSelected'
import { Sidebar } from './Sidebar'
import { NoteScreen } from '../notes/NoteScreen'
import { useSelector } from 'react-redux'

export const JournalScreen = () => {
    //obtenemos el active de las notes
    const {active} = useSelector(state => state.notes)


    return (
        
        
       
        <div className='journal__main-content animate__animated animate__fadeIn animate__faster'>
         <Sidebar/>
           

            
            <main>
                {
                    //verificamos si hay una nota activa en el caso de que no este, se muestra el componente nothinScreen
                    (active)
                    ? <NoteScreen/>
                    : <NothingSelected/>
                }
               
            </main>
            
        </div>
        
        
    )
}
