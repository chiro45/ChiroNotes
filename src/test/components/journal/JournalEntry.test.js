import { JournalEntry } from "../../../components/journal/JournalEntry";


import {mount} from 'enzyme'
import { Provider } from "react-redux";

import thunk from "redux-thunk";

import configureStore from 'redux-mock-store';
import { activeNote } from "../../../actions/notes";


const middlewares = [thunk];
const mockStore = configureStore(middlewares);




const initState = {
    
};

let store = mockStore(initState);
store.dispatch= jest.fn();

const nota  = {
    id: '1',
    title:'hola',
    body:'mundo',
    date: 0,
    url: 'https//chirolardo'
}


describe('Pruebas en el <JournalEntry/>', () => {
    const wrapper = mount(
        <Provider store={store}>
            
                  <JournalEntry {...nota}/>
            
        </Provider>
)
    test('debe de mostrarse correctamente', () => {

        expect(wrapper).toMatchSnapshot()
    });
   
    
    test('cuando hago click en la nota dispare la nota activa', () => {
      wrapper.find('.journal__entry').prop('onClick')();


      expect(store.dispatch).toHaveBeenCalledWith(
          activeNote(nota.id, {...nota}))
    });
    
});
