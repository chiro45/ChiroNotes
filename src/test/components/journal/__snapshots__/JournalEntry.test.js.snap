// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <JournalEntry/> debe de mostrarse correctamente 1`] = `
<div
  className="journal__entry pointer animate__animated animate__fadeInRight animate__fast"
  onClick={[Function]}
>
  <div
    className="journal__entry-picture"
    style={
      Object {
        "backgroundImage": "url(https//chirolardo)",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      hola
    </p>
    <p
      className="journal__entry-content"
    >
      mundo
    </p>
  </div>
  <div
    className="journal__entry-date-box"
  >
    <span>
      miércoles
    </span>
    <h4>
      31º
    </h4>
  </div>
</div>
`;
